apps/web-giddh/src/app/settings/integration/payment/icici/account-create-edit/account-create-edit.component.ts

```diff
@@ -0,0 +1,266 @@
+            });
+        } else {
+            this.accountForm = this.formBuilder.group({
+                bank: ['ICICI'],
+                loginId: ['', Validators.required],
+                accountNumber: ['', Validators.compose([Validators.required, Validators.minLength(9), Validators.maxLength(11)])],
+                accountUniqueName: ['', Validators.required],
+                paymentAlerts: [''],
+                userUniqueName: ['', Validators.required],
+                duration: [UNLIMITED_LIMIT],
+                maxAmount: ['']
+            });
+        }
+
+        this.loadUsersWithCompanyPermissions();
+        this.loadDefaultBankAccountsSuggestions();
+    }
+
+    /**
+     * Releases all the observables to avoid memory leaks
+     *
+     * @memberof AccountCreateEditComponent
+     */
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+    }
+
+    /**
+     * Loads all bank accounts
+     *
+     * @private
+     * @memberof AccountCreateEditComponent
+     */
+    private loadDefaultBankAccountsSuggestions(): void {
+        this.salesService.getAccountsWithCurrency('bankaccounts').pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.body?.results) {
+                const bankAccounts = response.body.results.map(account => ({
+                    label: account.name,
+                    value: account.uniqueName
+                }))
+                this.bankAccounts$ = observableOf(bankAccounts);
+            }
+            this.isLoading = false;
+        });
+    }
+
+    /**
+     * Loads users with company permissions
+     *
+     * @private
+     * @memberof AccountCreateEditComponent
+     */
+    private loadUsersWithCompanyPermissions(): void {
+        this.store.pipe(select(state => state.settings.usersWithCompanyPermissions), takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.usersList = [];
+                this.paymentAlertsUsersList = [];
+                let index = 0;
+
+                this.paymentAlertsUsersList.push({ index: index, label: this.commonLocaleData?.app_select_all, value: this.selectAllRecords });
+                index++;
+
+                response.forEach(user => {
+                    this.paymentAlertsUsersList.push({ index: index, label: user.userName, value: user.userUniqueName });
+                    this.usersList.push({ index: index, label: user.userName, value: user.userUniqueName });
+                    index++;
+                });
+            }
+        });
+    }
+
+    /**
+     * This will save the new account
+     *
+     * @memberof AccountCreateEditComponent
+     */
+    public saveNewAccount(): void {
+        if (!this.accountForm?.invalid) {
+            this.accountForm.get('paymentAlerts')?.patchValue(this.paymentAlerts.filter(user => user !== this.selectAllRecords).map(user => user));
```

Comment:
Please check `map` could be avoided as it is plainly returning the same data obtained from `filter`